src/canvas.py

```diff
@@ -242,13 +241,18 @@ def set_legend(self):
         """Set the legend of the graph"""
         if self.parent.plot_settings.legend:
             self.legends = []
-            lines, labels = self.ax.get_legend_handles_labels()
+            lines1, labels1 = self.ax.get_legend_handles_labels()
             lines2, labels2 = self.right_axis.get_legend_handles_labels()
             lines3, labels3 = self.top_left_axis.get_legend_handles_labels()
             lines4, labels4 = self.top_right_axis.get_legend_handles_labels()
-            self.top_right_axis.legend(
-                lines + lines2 + lines3 + lines4,
-                labels + labels2 + labels3 + labels4, loc=0, frameon=True)
+            new_lines = lines1 + lines2 + lines3 + lines4
+            new_labels = labels1 + labels2 + labels3 + labels4
+            labels = []
+            for label in new_labels:
+                labels.append(utilities.shorten_label(label))
```

Comment:
Very long labels make the plot unworkable, so it may be useful to shorten the label in the legend as well (like in this code), but 20 in the legend seems a bit low. In the end, this should maybe also be optional in the settings. 

This length of 20 works perfect for the item list in the panel). As a quick solution, I'd suggest having the max length as an optional argument with default value 20 in the utilities function, and then passing e.g. 40 for the max length in this piece here.

Reply:
Agreed, this seems like a scalable solution